korman/nodes/node_avatar.py

```diff
@@ -71,3 +71,388 @@ def export(self, exporter, bo, so):
+        row.prop(self, "loop_option")
+        if self.loop_option == "kLoop":
+            row = col.row()
+            row.prop(self, "num_loops")
+        col.separator()
+
+    @property
+    def stage_id(self):
+        idx = None
+        if self.find_output_socket("stage").is_linked:
+            parent_MSB = self.find_output_socket("stage").links[0].to_node
+            idx = next((idx for idx, socket in enumerate(parent_MSB.find_input_sockets("stage_refs")) if socket.is_linked and socket.links[0].from_node == self))
+        return idx
+
+
+class PlasmaBehaviorSocket(PlasmaNodeSocketBase, bpy.types.NodeSocket):
+    bl_color = (0.348, 0.186, 0.349, 1.0)
+
+
+class PlasmaMultiStageBehaviorNode(PlasmaNodeBase, bpy.types.Node):
+    bl_category = "AVATAR"
+    bl_idname = "PlasmaMultiStageBehaviorNode"
+    bl_label = "Multistage Behavior"
+    bl_width_default = 200
+
+    pl_attrib = ("ptAttribBehavior")
+
+    freeze_phys = BoolProperty(name="Freeze Physical",
+                              description="Freeze physical at end",
+                              default=False)
+    reverse_control = BoolProperty(name="Reverse Controls",
+                              description="Reverse forward/back controls at end",
+                              default=False)
+
+    input_sockets = OrderedDict([
+        ("seek_target", {
+            "text": "Seek Target",
+            "type": "PlasmaSeekTargetSocketIn",
+            "valid_link_sockets": {"PlasmaSeekTargetSocketOut"},
+        }),
+        ("stage_refs", {
+            "text": "Stage",
+            "type": "PlasmaAnimStageRefSocket",
+            "valid_link_nodes": "PlasmaAnimStageNode",
+            "valid_link_sockets": "PlasmaAnimStageRefSocket",
+            "link_limit": 1,
+            "spawn_empty": True,
+        }),
+    ])
+
+    output_sockets = OrderedDict([
+        ("hosts", {
+            "text": "Host Script",
+            "type": "PlasmaBehaviorSocket",
+            "valid_link_nodes": {"PlasmaPythonFileNode"},
+            "spawn_empty": True,
+        })
+    ])
+
+    def draw_buttons(self, context, layout):
+        col = layout.column()
+        col.label("Options:")
+        col.prop(self, "freeze_phys")
+        col.prop(self, "reverse_control")
+
+    def get_key(self, exporter, so):
+        return self._find_create_key(plMultistageBehMod, exporter, so=so)
+
+    def export(self, exporter, bo, so):
+        seek_socket = self.find_input_socket("seek_target")
+
+        if seek_socket.is_linked:
+            smartSeek = True
+            seek_target = seek_socket.links[0].from_node.target
+        elif seek_socket.auto_target:
+            smartSeek = True
+            seek_target = bo
+        else:
+            smartSeek = False
+            seek_target = bo
+
+        # We need to make sure the object we're attached to is being exported as a Plasma Object, otherwise this is all nonsense
+        seek_target.plasma_object.enabled = True
```

Comment:
The exporter will proceed just fine without it being marked Plasma Object. The other modifiers on the object just won't be exported.

Reply:
Ah, I did not know this!  There was more logic involved here, before we made the decision to assume we're attached to the object we're going to export on, so it's a bit redundant besides, I suppose.